crates/exchange-rates/src/feeds.rs

```diff
@@ -0,0 +1,150 @@
+use std::{fs::File, io::BufReader, path::PathBuf, sync::Arc};
+
+use crate::{types::{ChainlinkFeedData, PythFeedData, PythId}, utils::get_asset_match};
+use ethers::types::{Address, I256};
+use ethui_abis::ChainlinkAgregatorV3;
+use ethui_networks::Networks;
+use ethui_types::GlobalState;
+
+pub async fn get_chainlink_price(base_asset: String, quote_asset: String) -> I256 {
+    let networks = Networks::read().await;
+    let network = networks.get_current();
+    let current_chain_id = network.chain_id;
+    let provider = network.get_provider();
+    let client = Arc::new(provider);
+
+    let filename = format!("{}.json", current_chain_id);
+    let base_path = PathBuf::from("../target/exchange-rates/chainlink/");
```

Comment:
this won't work. you're opening a file from a relative directory in runtime. but the binary will be shipped without the target directory, so this will fail unless you're running it with `cargo run ...`

the correct approach is to read the file at compile time and store it in a constant

for example: https://doc.rust-lang.org/std/macro.include_str.html